nixos/modules/services/continuous-integration/buildkite-agent.nix

```diff
@@ -65,27 +66,43 @@ in
         home = "/var/lib/buildkite-agent";
         createHome = true;
         description = "Buildkite agent user";
+        extraGroups = [ "keys" ];
       };
 
     environment.systemPackages = [ pkgs.buildkite-agent ];
 
     systemd.services.buildkite-agent =
+      let copyOrEcho       = x: target: perms:
+                             (if isPath x
+                              then "cp -f ${x} ${target}; "
+                              else "echo '${x}' > ${target}; ")
+                             + "${pkgs.coreutils}/bin/chmod ${toString perms} ${target}; ";
+          catOrLiteral     = x:
+                             (if isPath x
+                              then "$(cat ${toString x})"
+                              else "${x}");
+      in
       { description = "Buildkite Agent";
         wantedBy = [ "multi-user.target" ];
         after = [ "network.target" ];
         environment.HOME = "/var/lib/buildkite-agent";
         preStart = ''
             ${pkgs.coreutils}/bin/mkdir -m 0700 -p /var/lib/buildkite-agent/.ssh
+            ${copyOrEcho cfg.openssh.privateKey "/var/lib/buildkite-agent/.ssh/id_rsa"     600}
+            ${copyOrEcho cfg.openssh.publicKey  "/var/lib/buildkite-agent/.ssh/id_rsa.pub" 600}
 
-            echo "${cfg.openssh.privateKey}" > /var/lib/buildkite-agent/.ssh/id_rsa
-            ${pkgs.coreutils}/bin/chmod 600 /var/lib/buildkite-agent/.ssh/id_rsa
-
-            echo "${cfg.openssh.publicKey}" > /var/lib/buildkite-agent/.ssh/id_rsa.pub
-            ${pkgs.coreutils}/bin/chmod 600 /var/lib/buildkite-agent/.ssh/id_rsa.pub
+            cat > "/var/lib/buildkite-agent/buildkite-agent.cfg" <<EOF
```

Comment:
Missing a comment why this is done this way, to prevent issues in the future

Reply:
Done.